### PR TITLE
Create markup orders

### PR DESCRIPTION
**Pull request by Pedrael**

### CODE REVIEW

File: src/components/ProductCard.tsx [hunk 1]
```diff
@@ -27,11 +29,13 @@ export const ProductCard = ({
   const { t } = useTranslation()
   const [open, setOpen] = useState(false)
   const dispatch = useDispatch()
-  const orderName = useSelector(
-    (state: RootState) =>
-      state.order.value.find((order) => order.products.find((id) => id))
-        ?.title ?? 'none',
+  const orderNameSelector = createSelector(
```
Comment (ReidenXerx) [inline src/components/ProductCard.tsx]: All selectors should be in separate module

File: src/components/orderCard.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  products,
+  ...props
+}: OrderProps) => {
+  const { t } = useTranslation()
+  const [open, setOpen] = useState(false)
+  const [visible, isVisible] = useState(false)
+  const dispatch = useDispatch()
+  const productsByIdSelector = createSelector(
+    ({ product: { value } }: RootState) => value,
+    (value) => value.filter((product) => products.includes(product.id)),
```
Comment (ReidenXerx) [inline src/components/orderCard.tsx]: Naming

File: src/components/orderCard.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  }
+  const handleAccept = (id: number) => {
+    setOpen(false)
+    dispatch(removeOrderActionById(id))
+  }
+  const handleVisible = () => {
+    isVisible(!visible)
+  }
+  const prices = productsById.map(({ price }) => price)
+  const sumPrices = (prices: Array<Array<Price>>) => {
```
Comment (ReidenXerx) [inline src/components/orderCard.tsx]: .reduce((acc, {cur, val}) => {
Return {...acc, cur: acc[cur] + val}
}, {cur1, cur2} )

File: src/components/orderCard.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        borderRadius="8px"
+        border="1px solid"
+        borderColor={gray}
+        gap="20px"
+        {...props}
+      >
+        <Typography>{title}</Typography>
+        <Typography>{date.toLocaleDateString()}</Typography>
+        <Stack>
+          {sumPrices(prices).map(
```
Comment (ReidenXerx) [inline src/components/orderCard.tsx]: By conventions map should return value on **every** iteration

File: src/pages/index/index.page.tsx [hunk 5]
```diff
@@ -1,7 +1,26 @@
+import { useSelector } from 'react-redux'
+import { RootState } from '../../store'
+import { OrderCard, OrderProps } from '../../components/OrderCard'
+import { Container, Typography } from '@mui/material'
+import { cream } from '../../constants'
+import { useTranslation } from 'react-i18next'
+import { createSelector } from '@reduxjs/toolkit'
+
 export const Page = () => {
+  const ordersSelector = createSelector(
```
Comment (ReidenXerx) [inline src/pages/index/index.page.tsx]: Same

File: src/pages/products/index.page.tsx [hunk 6]
```diff
@@ -11,14 +11,19 @@ import { ProductCard, ProductProps } from '../../components/ProductCard'
...
+import { createSelector } from '@reduxjs/toolkit'
 import { RootState } from '../../store'
 import { productTypes } from '../../types'
 import { useTranslation } from 'react-i18next'
 
 export const Page = () => {
   const { t } = useTranslation()
   const [type, setType] = useState<productTypes>(productTypes.All)
-  const products = useSelector((state: RootState) => state.product.value)
+  const productsSelector = createSelector(
```
Comment (ReidenXerx) [inline src/pages/products/index.page.tsx]: Same

File: src/pages/products/index.page.tsx [hunk 7]
```diff
@@ -11,14 +11,19 @@ import { ProductCard, ProductProps } from '../../components/ProductCard'
...
+    ({ product: { value } }: RootState) => value,
+    (value) => value,
```
Comment (ReidenXerx) [inline src/pages/products/index.page.tsx]: Worst practice 
Pls read docs about output selector

File: src/pages/index/index.page.tsx [hunk 8]
```diff
@@ -1,7 +1,26 @@
...
+    ({ order: { value } }: RootState) => value,
+    (value) => value,
```
Comment (ReidenXerx) [inline src/pages/index/index.page.tsx]: Same

File: src/types.ts [hunk 9]
```diff
@@ -31,7 +31,7 @@ export enum productTypes {
 
 export type Product = {
   id: number
-  serialNumber: number
```
Comment (ReidenXerx) [inline src/types.ts:34]: ??😂

**Review by ReidenXerx — CHANGES_REQUESTED** [review]

File: src/types.ts (hunk 9 above)
Comment (Pedrael) [reply src/types.ts:34]: I swapped to serialnumber (with small letter) for react warning fix

File: src/components/ProductCard.tsx (hunk 1 above)
Comment (Pedrael) [reply src/components/ProductCard.tsx]: fixed

File: src/components/orderCard.tsx (hunk 2 above)
Comment (Pedrael) [reply src/components/orderCard.tsx]: fixed

File: src/pages/index/index.page.tsx (hunk 5 above)
Comment (Pedrael) [reply src/pages/index/index.page.tsx]: fixed

File: src/components/orderCard.tsx (hunk 4 above)
Comment (Pedrael) [reply src/components/orderCard.tsx]: fixed

File: src/pages/products/index.page.tsx (hunk 6 above)
Comment (Pedrael) [reply src/pages/products/index.page.tsx]: fixed

File: src/pages/products/index.page.tsx (hunk 7 above)
Comment (Pedrael) [reply src/pages/products/index.page.tsx]: fixed

File: src/pages/index/index.page.tsx (hunk 8 above)
Comment (Pedrael) [reply src/pages/index/index.page.tsx]: fixed

File: src/components/orderCard.tsx (hunk 3 above)
Comment (Pedrael) [reply src/components/orderCard.tsx]: fixed

**Review by ReidenXerx — APPROVED** [review]

**Pull request merged**